lib/propolis/src/hw/bhyve/rtc.rs

```diff
@@ -94,7 +94,14 @@ impl BhyveRtc {
 
         // High-mem
         if high_mem > 0 {
-            let high = ((high_mem / CHUNK) as u32).to_le_bytes();
+            // If high_mem is 1TiB or larger, the division below produces a
+            // number that overflows the 24 bits available at
+            // `CMOS_OFF_MEM_HIGH`. Clamp the value so guests aren't subjected
+            // to arbitrary wrapping. OVMF is told about the highmem layout via
+            // E820 table anyway, so the only thing that might care about these
+            // bytes are guest OSes that check the RTC CMOS bytes directly.
```

Comment:
Maybe include mention that this clamping fails at 256TiB (where we roll over the 2^32 * 64KiB)?

Reply:
moved the peas around the plate instead (clamp to u32::MAX before converting to u32 so the only wrapping here is when highmem is more than 64b)